common/sw-update/dev-updates-profile.cpp

```diff
@@ -67,20 +67,29 @@ namespace rs2
             versions_db_manager::component_part_type part,
             dev_updates_profile::update_description& result)
         {
-            versions_db_manager::version required_version;
-            bool query_ok = up_handler.query_versions(dev_name, part, policy, required_version);
-            if (query_ok)
+            if (_keep_trying)
             {
-                up_handler.get_version_download_link(part, required_version, result.download_link);
-                up_handler.get_version_release_notes(part, required_version, result.release_page);
-                up_handler.get_version_description(part, required_version, result.description);
-                result.ver = required_version;
+                versions_db_manager::version required_version;
+                auto query_status = up_handler.query_versions(dev_name, part, policy, required_version);
+                if (query_status == versions_db_manager::VERSION_FOUND)
+                {
+                    up_handler.get_version_download_link(part, required_version, result.download_link);
+                    up_handler.get_version_release_notes(part, required_version, result.release_page);
+                    up_handler.get_version_description(part, required_version, result.description);
+                    result.ver = required_version;
+
+                    std::stringstream ss;
+                    ss << std::string(result.ver) << " (" << up_handler.to_string(policy) << ")";
+                    result.name = ss.str();
+                    return true;
+                }
+                else if (query_status == versions_db_manager::DB_LOAD_FAILURE)
+                {
+                    _keep_trying = false;
```

Comment:
`// once we fail to connect, we don't want to try again -- each try is a 5-second maximum wait (e.g., when we have internet but no proxy defined) and we have several calls to try_parse_update(), meaning we could potentially wait even half a minute!`